threads.go

```diff
@@ -271,22 +279,24 @@ func (t *threads) AddThread(
 	return t.store.ThreadInfo(id)
 }
 
-// PullThread for new records.
-// Logs owned by this host are traversed locally.
-// Remotely addressed logs are pulled from the network.
-func (t *threads) PullThread(ctx context.Context, id thread.ID) error {
-	log.Debugf("pulling thread %s...", id.String())
+func (t *threads) getThreadSemaphore(id thread.ID) chan struct{} {
 	var ptl chan struct{}
 	var ok bool
-	// ToDo: fix concurrency
 	t.pullLock.Lock()
 	defer t.pullLock.Unlock()
 	if ptl, ok = t.pullLocks[id]; !ok {
 		ptl = make(chan struct{}, 1)
 		t.pullLocks[id] = ptl
 	}
-	// t.pullLock.Unlock()
+	return ptl
+}
```

Comment:
Some helper function to get the semaphore for a thread.
Will be more useful for later changes that keep fixing concurrency per thread.